pinot-query-planner/src/test/resources/queries/GroupBy.json

```diff
@@ -0,0 +1,78 @@
+{
+  "group_by_planning_tests": {
+    "queries": [
+      {
+        "description": "Group by with select and aggregate column",
+        "sql": "EXPLAIN PLAN FOR SELECT a.col1, SUM(a.col3) FROM a GROUP BY a.col1",
+        "output": [
+          "Execution Plan",
+          "\nLogicalAggregate(group=[{0}], EXPR$1=[$SUM0($1)])",
+          "\n  LogicalExchange(distribution=[hash[0]])",
+          "\n    LogicalAggregate(group=[{2}], EXPR$1=[$SUM0($1)])",
+          "\n      LogicalTableScan(table=[[a]])",
+          "\n"
+        ]
+      },
+      {
+        "description": "Group by with filter",
+        "sql": "EXPLAIN PLAN FOR SELECT a.col1, SUM(a.col3) FROM a WHERE a.col3 >= 0 AND a.col2 = 'a' GROUP BY a.col1",
+        "output": [
+          "Execution Plan",
+          "\nLogicalAggregate(group=[{0}], EXPR$1=[$SUM0($1)])",
+          "\n  LogicalExchange(distribution=[hash[0]])",
+          "\n    LogicalAggregate(group=[{2}], EXPR$1=[$SUM0($1)])",
+          "\n      LogicalProject(col2=[$0], col3=[$1], col1=[$2])",
+          "\n        LogicalFilter(condition=[AND(>=($1, 0), =($0, 'a'))])",
+          "\n          LogicalTableScan(table=[[a]])",
+          "\n"
+        ]
+      },
+      {
+        "description": "Group by count(*) with filter",
+        "sql": "EXPLAIN PLAN FOR SELECT a.col1, COUNT(*) FROM a WHERE a.col3 >= 0 AND a.col2 = 'a' GROUP BY a.col1",
+        "output": [
+          "Execution Plan",
+          "\nLogicalAggregate(group=[{0}], EXPR$1=[$SUM0($1)])",
+          "\n  LogicalExchange(distribution=[hash[0]])",
+          "\n    LogicalAggregate(group=[{2}], EXPR$1=[COUNT()])",
+          "\n      LogicalProject(col2=[$0], col3=[$1], col1=[$2])",
```

Comment:
this one looks incorrect, project should only keep col1 b/c col2 and col3 already be pushed past into the filter . could you add a TODO comment in the JSON indicating that this needs follow up?

Reply:
done - left a note